controlplane/kubeadm/internal/machine_collection.go

```diff
@@ -56,14 +56,21 @@ func NewFilterableMachineCollectionFromMachineList(machineList *clusterv1.Machin
 }
 
 // Insert adds items to the set.
-func (s FilterableMachineCollection) Insert(machines ...*clusterv1.Machine) FilterableMachineCollection {
+func (s FilterableMachineCollection) Insert(machines ...*clusterv1.Machine) {
```

Comment:
This is just some cleanup, while implementing `Sub` I noticed that we had both mutating and copying methods on `FilterableMachineCollection`. `Insert` was mutating but by returning a collection it implies that it's copying. We weren't using the return value anywhere except for some tests, and even there it was only being used to assign it to the same variable that was being mutated.